akka-stream/src/main/scala/akka/stream/impl/io/InputStreamPublisher.scala

```diff
@@ -47,7 +47,7 @@ private[akka] class InputStreamPublisher(is: InputStream, bytesReadPromise: Prom
   def readAndSignal(): Unit =
     if (isActive) {
       readAndEmit()
```

Comment:
I know (it would be nice to have in the title btw), I just meant that oftentimes mutable state is the Devil.